db/migrate/20241112093940_set_eu_building_stock_values.rb

```diff
@@ -0,0 +1,153 @@
+    'households_number_of_detached_houses_2005_present' => 'present_number_of_detached_houses_2005_present',
+    'households_number_of_semi_detached_houses_before_1945' => 'present_number_of_detached_houses_before_1945',
+    'households_number_of_semi_detached_houses_1945_1964' => 'present_number_of_detached_houses_1945_1964',
+    'households_number_of_semi_detached_houses_1965_1984' => 'present_number_of_detached_houses_1965_1984',
+    'households_number_of_semi_detached_houses_1985_2004' => 'present_number_of_detached_houses_1985_2004',
+    'households_number_of_semi_detached_houses_2005_present' => 'present_number_of_detached_houses_2005_present'
+  }.freeze
+
+  # Inputs and corresponding area attribute
+  EXISTING_STOCK_MAPPING = {
+    'households_number_of_apartments_before_1945' => 'present_number_of_apartments_before_1945',
+    'households_number_of_apartments_1945_1964' => 'present_number_of_apartments_1945_1964',
+    'households_number_of_apartments_1965_1984' => 'present_number_of_apartments_1965_1984',
+    'households_number_of_apartments_1985_2004' => 'present_number_of_apartments_1985_2004',
+    'households_number_of_apartments_2005_present' => 'present_number_of_apartments_2005_present',
+    'households_number_of_terraced_houses_before_1945' => 'present_number_of_terraced_houses_before_1945',
+    'households_number_of_terraced_houses_1945_1964' => 'present_number_of_terraced_houses_1945_1964',
+    'households_number_of_terraced_houses_1965_1984' => 'present_number_of_terraced_houses_1965_1984',
+    'households_number_of_terraced_houses_1985_2004' => 'present_number_of_terraced_houses_1985_2004',
+    'households_number_of_terraced_houses_2005_present' => 'present_number_of_terraced_houses_2005_present'
+  }.freeze
+
+
+  def up
+    @defaults = JSON.load(File.read(
+      Rails.root.join("db/migrate/#{File.basename(__FILE__, '.rb')}/dataset_values.json")
+    ))
+
+    migrate_scenarios do |scenario|
+
+      # Filter scenarios for the relevant country datasets
+      next unless Atlas::Dataset.exists?(scenario.area_code) && COUNTRIES.include?(scenario.area_code)
+
+      set_existing_stock_excl_detached_semi_detached(scenario)
+      set_existing_stock_detached_semi_detached(scenario)
+
+    end
+  end
+
+  # The numbers of existing buildings should be set to the default value if not set in the scenario.
+  def set_existing_stock_excl_detached_semi_detached(scenario)
+    # Load scenario defaults fron json file
+    scenario_defaults = @defaults[scenario.area_code.to_s]
+    unless scenario_defaults
+      return
+    end
+
+    EXISTING_STOCK_MAPPING.each do |key, area|
+      # Obtain default area values
+      value = scenario_defaults[area]
+      if value.nil?
+        next
+      end
+      # If slider not set, set slider to the dataset default value
+      unless scenario.user_values.key?(key)
+        scenario.user_values[key] = value
+      end
+    end
+  end
```

Comment:
Isn't this part redundant? If the slider isn't set originally, then the slider just has the start value. If we then update the building stock, the start value changes and we don't have to set it explicitly right?